client/src/App.js

```diff
@@ -1,12 +1,32 @@
 import { Route, Routes } from "react-router-dom";
+import { AppContext } from "./AppContext";
+import { useState } from "react";
 import "./App.css";
 
 import Home from "./pages/Home";
+import Result from "./pages/Result" ;
 
-const App = () => (
-	<Routes>
+const App = () => {
+	const [userName ,setUserName] = useState("");
+	const [codewarsData, setCodewarsData] = useState([]) ;
+	const [githubData, setGitHubData] = useState([]);
+	return(
+	<AppContext.Provider value={
+		{
+			userName ,
+			setUserName ,
+			codewarsData ,
+			setCodewarsData ,
+			githubData ,
+			setGitHubData
+		}
+	}>
```

Comment:
This looks good, we can have a look at extracting it in future PRs to clean up this file but this is good for now